server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -8208,18 +8208,22 @@ private void postProcessingUnmanageVM(UserVmVO vm) {
         Long cpu = offering.getCpu() != null ? new Long(offering.getCpu()) : 0L;
         Long ram = offering.getRamSize() != null ? new Long(offering.getRamSize()) : 0L;
         // First generate a VM stop event if the VM was not stopped already
+        boolean resourceCountNotDecremented = true;
         if (vm.getState() != State.Stopped) {
             UsageEventUtils.publishUsageEvent(EventTypes.EVENT_VM_STOP, vm.getAccountId(), vm.getDataCenterId(),
                     vm.getId(), vm.getHostName(), vm.getServiceOfferingId(), vm.getTemplateId(),
                     vm.getHypervisorType().toString(), VirtualMachine.class.getName(), vm.getUuid(), vm.isDisplayVm());
             resourceCountDecrement(vm.getAccountId(), vm.isDisplayVm(), cpu, ram);
+            resourceCountNotDecremented = false;
```

Comment:
why not postpone decrementing the count altogether and only do it at the destroy call below? It seems that one is being done in any case. Only the removal of `resourceCountDecrement(vm.getAccountId(), vm.isDisplayVm(), cpu, ram);` is needed.